src/Akeneo/Tool/Bundle/ApiBundle/Cache/WarmupQueryCache.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Tool\Bundle\ApiBundle\Cache;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Warms up LRU cached queries in order to minimize the number of SQL queries performed towards the DB.
```

Comment:
I propose this to expose the drawbacks:

```suggestion
 * When importing product/product model, a lot of Database queries are triggered. We suffer of the 1+n queries problem which drastically impact the performance.
 * @ see https://www.sitepoint.com/silver-bullet-n1-problem/
 * 
 * One way to avoid that is to pre-load the need data in 1 query instead of 1 +n queries. This trick increase drastically the performance. There are two drawbacks though:
 * - it means we know the needed subrequests in sub-layers, meaning that we have knowledge of an internal behavior. Therefore, changing the implementation of the internal behavior can make this cache warmup obsolete.
 
Our abstraction to import products is leaky: https://www.joelonsoftware.com/2002/11/11/the-law-of-leaky-abstractions/
 
 * - the queries are cached, and so stateful. Do note that this was already the case even before this class, which mitigate this drawback
 
 Early analyzis of the initial request allows to gather the queries' arguments (e.g attribute codes).
```